src/main/java/org/springframework/boot/autoconfigure/data/jdbc/JdbcRepositoriesMyBatisAutoConfiguration.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.boot.autoconfigure.data.jdbc;
+
+import org.apache.ibatis.session.SqlSessionFactory;
+import org.mybatis.spring.boot.autoconfigure.MybatisAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.data.jdbc.core.DataAccessStrategy;
+import org.springframework.data.jdbc.core.DefaultDataAccessStrategy;
+import org.springframework.data.jdbc.core.SqlGeneratorSource;
+import org.springframework.data.jdbc.mapping.model.JdbcMappingContext;
+import org.springframework.data.jdbc.mybatis.MyBatisDataAccessStrategy;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcOperations;
+
+/**
+ * Autoconfigure Spring Data JDBC
+ *
+ * @author Jens Schauder
+ * @author Greg Turnquist
+ */
+@Configuration
+@ConditionalOnClass(SqlSessionFactory.class)
+@Import(MybatisAutoConfiguration.class)
+public class JdbcRepositoriesMyBatisAutoConfiguration {
+
+	@Bean("dataAccessStrategy")
+	@ConditionalOnMissingBean
+	DataAccessStrategy dataAccessStrategyWithMyBatisSupport(NamedParameterJdbcOperations operations,
+			SqlGeneratorSource sqlGeneratorSource, JdbcMappingContext context, SqlSessionFactory sqlSessionFactory) {
+		System.out.println("configuring mybatis");
+		return JdbcRepositoriesAutoConfiguration.buildDataAccessStrategy(new MyBatisDataAccessStrategy(sqlSessionFactory),
+				new DefaultDataAccessStrategy(sqlGeneratorSource, operations, context));
+	}
```

Comment:
If we're using MyBatis, why do we need the other one? Is this in the event you only use MyBatis for perhaps one particular aggregate?

Reply:
With MyBatis statements may come from the `SqlSessionFactory`. If it doesn't contain the query in question we still fall back on the `DefaultDataAccessStrategy`